packages/flutter/lib/src/material/drawer.dart

```diff
@@ -676,18 +674,12 @@ class DrawerControllerState extends State<DrawerController> with SingleTickerPro
         isDesktop = true;
     }
 
-    double? dragAreaWidth = widget.edgeDragWidth;
-    if (widget.edgeDragWidth == null) {
-      final EdgeInsets padding = MediaQuery.paddingOf(context);
-      switch (textDirection) {
-        case TextDirection.ltr:
-          dragAreaWidth = _kEdgeDragWidth +
-            (drawerIsStart ? padding.left : padding.right);
-        case TextDirection.rtl:
-          dragAreaWidth = _kEdgeDragWidth +
-            (drawerIsStart ? padding.right : padding.left);
-      }
-    }
+    final double dragAreaWidth = widget.edgeDragWidth ?? _kEdgeDragWidth + switch ((widget.alignment, Directionality.of(context))) {
```

Comment:
Sounds good! I made whitespace changes for readability; we could break it into multiple lines (but it would no longer be `final` and non-nullable):

```dart
double? dragAreaWidth = widget.edgeDragWidth;
dragAreaWidth ??= _kEdgeDragWidth + switch ((widget.alignment, Directionality.of(context))) {
  (DrawerAlignment.start, TextDirection.ltr) => MediaQuery.paddingOf(context).left,
  (DrawerAlignment.start, TextDirection.rtl) => MediaQuery.paddingOf(context).right,
  (DrawerAlignment.end,   TextDirection.rtl) => MediaQuery.paddingOf(context).left,
  (DrawerAlignment.end,   TextDirection.ltr) => MediaQuery.paddingOf(context).right,
};
```

Reply:
Gotcha. Feel free to keep it as is then :)